client/go/outline/router/ip_router.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package router
+
+import (
+	"container/list"
+	"fmt"
+	"net/netip"
+)
+
+// Compile-time check
+var _ Router[netip.Prefix, netip.Addr, any] = (*IPRouter[any])(nil)
+
+// "V" is typically expected to be a dialer of some kind
+type IPRouter[V any] struct {
```

Comment:
Call it IPMap?
It's easier to reason about.
I'd rename the methods to Set and Get, like a map.

Reply:
Also, this is not a router. It's not routing anything. Perhaps a RoutingTable?